install/src/command.rs

```diff
@@ -786,44 +795,59 @@ pub fn update(config_file: &str) -> Result<bool, String> {
             ExplicitRelease::Semver(release_semver) => {
                 let download_url = github_release_download_url(release_semver);
                 let release_dir = config.release_dir(&release_semver);
-                let download_url = if release_dir.join(".ok").exists() {
+                let download_url = if release_dir.exists() {
                     // If this release_semver has already been successfully downloaded, no update
                     // needed
-                    println!("{} is present, no download required.", release_semver);
+                    println!("{} found in cache", release_semver);
                     None
                 } else {
                     Some(download_url)
                 };
                 (download_url, release_dir)
             }
             ExplicitRelease::Channel(release_channel) => {
-                let release_dir = config.release_dir(&release_channel);
+                let version_url = release_channel_version_url(release_channel);
+
+                let (_temp_dir, temp_file, _temp_archive_sha256) =
+                    download_to_temp(&version_url, None)
+                        .map_err(|err| format!("Unable to download {}: {}", version_url, err))?;
+
+                let update_release_version = load_release_version(&temp_file)?;
+
+                let release_id = format!("{}-{}", release_channel, update_release_version.commit);
```

Comment:
Stop reusing the channel name for the install directory.  Instead use "channel-commit".  In addition to avoiding the ^C issue, this gives us a backdoor to revert to an older channel build if you just upgraded to broken one (by messing around manually in `~/.local/share/solana/install/`.   The old/new commit is also printed to help with this